docs/details/widgets/chart.rst

```diff
@@ -4,110 +4,241 @@
-static or dynamically allocated. Note: you should call
-:cpp:expr:`lv_chart_refresh(chart)` after the external data source has been
-updated to update the chart.
+If you wish to have the chart use your own Y-value array instead of the one provided,
+you can do so with
 
-The value array of a series can be obtained with
-:cpp:expr:`lv_chart_get_y_array(chart, ser)`, which can be used with
-``ext_array`` or *normal array*\ s.
+    :cpp:expr:`lv_chart_set_ext_y_array(chart, series, value_array)`.
 
-For :cpp:enumerator:`LV_CHART_TYPE_SCATTER` type
-:cpp:expr:`lv_chart_set_ext_x_array(chart, ser, value_array)` and
-:cpp:expr:`lv_chart_get_x_array(chart, ser)` can be used as well.
+You are responsible for ensuring the array you provide contains at least
+``chart->point_cnt`` elements in it.
 
-Modify the data
----------------
+``value_array`` should look like this: ``int32_t * value_array[num_points]``.  Only
+the array's pointer is saved in the series so its contents need to remain available
+for the life of the series, i.e. the array needs to be global, static or dynamically
+allocated.
 
-You have several options to set the data of series:
+.. note::
+    Call :cpp:expr:`lv_chart_refresh(chart)` when a chart's data has changed to
+    signal that the chart should be re-rendered next time a display refresh occurs.
+    You do not need to do this if you are using the provided value array(s) and
+    setting values with ``lv_chart_set_...value_...()`` functions.  See below
+    for more information about these functions.
 
-1. Set the values manually in the array like ``ser1->points[3] = 7`` and refresh the chart with :cpp:enumerator:`lv_chart_refresh(chart)`.
-2. Use :cpp:expr:`lv_chart_set_value_by_id(chart, ser, id, value)` where ``id`` is the index of the point you wish to update.
-3. Use the :cpp:expr:`lv_chart_set_next_value(chart, ser, value)`.
-4. Initialize all points to a given value with :cpp:expr:`lv_chart_set_all_value(chart, ser, value)`.
+A pointer to the Y-value array of a series can be obtained with
+:cpp:expr:`lv_chart_get_y_array(chart, series)`.  This is true whether you are using
+the provided Y-value array or provided your own.
 
-Use :cpp:enumerator:`LV_CHART_POINT_NONE` as value to make the library skip drawing
-that point, column, or line segment.
+For SCATTER-type charts,
 
-For :cpp:enumerator:`LV_CHART_TYPE_SCATTER` type
-:cpp:expr:`lv_chart_set_value_by_id2(chart, ser, id, value)` and
-:cpp:expr:`lv_chart_set_next_value2(chart, ser, x_value, y_value)` can be used
-as well.
+- :cpp:expr:`lv_chart_set_ext_x_array(chart, series, value_array)` and
+- :cpp:expr:`lv_chart_get_x_array(chart, series)`
+
+can be used as well.
+
+Modifying data
+--------------
+
+You have several options to set the Y-values for a data series:
+
+1. Set the values programmatically in the array like ``ser1->points[3] = 7`` and refresh the
+   chart with :cpp:enumerator:`lv_chart_refresh(chart)`.
```

Comment:
I only just spotted this while using your changes as a reference for writing new docs :smile:

If you edit this file again in another PR and you remember this comment, please make this ":cpp:expr:\`lv_chart_refresh(chart)\`"

Reply:
> I only just spotted this while using your changes as a reference for writing new docs 😄

Oh my goodness!  **Done.**  I have quite a few "minor" edits in docs already submitted that I am planning to submit as a "wrap-up" once I get through all the Widgets.